lib/AST/PluginLoader.cpp

```diff
@@ -46,69 +48,59 @@ PluginRegistry *PluginLoader::getRegistry() {
   return Registry;
 }
 
-llvm::Optional<std::string>
-PluginLoader::lookupExplicitLibraryPluginByModuleName(Identifier moduleName) {
-  // Look for 'lib${module name}(.dylib|.so)'.
-  SmallString<64> expectedBasename;
-  expectedBasename.append("lib");
-  expectedBasename.append(moduleName.str());
-  expectedBasename.append(LTDL_SHLIB_EXT);
-
-  // Try '-load-plugin-library'.
-  for (const auto &libPath :
-       Ctx.SearchPathOpts.getCompilerPluginLibraryPaths()) {
-    if (llvm::sys::path::filename(libPath) == expectedBasename) {
-      return libPath;
-    }
-  }
-  return None;
-}
+std::pair<std::string, std::string>
+PluginLoader::lookupPluginByModuleName(Identifier moduleName) {
+  auto fs = Ctx.SourceMgr.getFileSystem();
 
-llvm::Optional<std::string>
-PluginLoader::lookupLibraryPluginInSearchPathByModuleName(
-    Identifier moduleName) {
   // Look for 'lib${module name}(.dylib|.so)'.
-  SmallString<64> expectedBasename;
-  expectedBasename.append("lib");
-  expectedBasename.append(moduleName.str());
-  expectedBasename.append(LTDL_SHLIB_EXT);
-
-  // Try '-plugin-path'.
-  auto fs = Ctx.SourceMgr.getFileSystem();
-  for (const auto &searchPath : Ctx.SearchPathOpts.PluginSearchPaths) {
-    SmallString<128> fullPath(searchPath);
-    llvm::sys::path::append(fullPath, expectedBasename);
-    if (fs->exists(fullPath)) {
-      return std::string(fullPath);
+  // FIXME: Shared library prefix might be different between platforms.
+  SmallString<64> pluginLibBasename;
+  pluginLibBasename.append("lib");
+  pluginLibBasename.append(moduleName.str());
+  pluginLibBasename.append(LTDL_SHLIB_EXT);
+
+  // FIXME: Should we create a lookup table keyed by module name?
```

Comment:
Currently we create a lookup table for `-load-plugin-executable` only. But we could lazily create a lookup table for all keyed by module names. But that's need to scan (iterate directory entries) all search paths, so I'm not sure it worth it.